packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -10,11 +11,18 @@ function findExampleByKey(httpContent: IHttpContent, exampleKey: string) {
   return httpContent.examples && httpContent.examples.find(example => example.key === exampleKey);
 }
 
-function findHttpContentByMediaType(
+function findBestHttpContentByMediaType(
   response: IHttpOperationResponse,
   mediaType: string,
 ): IMediaTypeContent | undefined {
-  return response.contents.find(content => content.mediaType === mediaType);
+  return (
+    response.contents.find(content => content.mediaType === mediaType) ||
+    response.contents.filter(content => content.mediaType !== '*').find(content => {
+      const parsedRequestedMediaType = mediaTypeParser.fromString(mediaType);
+      const parsedCandidateMediaType = mediaTypeParser.fromString(content.mediaType);
+      return parsedRequestedMediaType.suffix === parsedCandidateMediaType.subtype;
```

Comment:
This looks annoying. 

When give `text/plain` the property `media.subtype` is `'plain'`, but when given `'application/vnd.company.app.entity-v2+xml` the `media.subtype` is `vnd.company.app.entity-v2` and xml is on `media.suffix`.

This means your code only match exact matches `content => content.mediaType === mediaType` or mismatched specifics when the client requests `application/foo+json` but only `application/json` is defined in the spec. 

If the user requested application/json and the spec had `application/json; charset=utf8` this would also fail. 

I think you want to find a content negotiation tool which will handle the comparisons for you, because this is a can of worms.

Reply:
I wish there would be a package handling all this stuff seriously — unfortunately I haven't been able to find a single one doing this. I'll give it another round…but no luck so far.